src/client/cli/cmd/exec.cpp

```diff
@@ -19,11 +19,32 @@
 #include "common_cli.h"
 
 #include <multipass/cli/argparser.h>
+#include <multipass/file_ops.h>
 #include <multipass/ssh/ssh_client.h>
 
 namespace mp = multipass;
 namespace cmd = multipass::cmd;
 
+namespace
+{
+const QString work_dir_option_name{"working-directory"};
+const QString no_dir_mapping_option{"no-map-working-directory"};
+
+bool is_dir_mounted(const QStringList& split_current_dir, const QStringList& split_source_dir)
+{
+    int source_dir_size = split_source_dir.size();
+
+    if (split_current_dir.size() < source_dir_size)
+        return false;
+
+    for (int i = 0; i < source_dir_size; ++i)
+        if (split_current_dir[i] != split_source_dir[i])
+            return false;
+
+    return true;
+}
```

Comment:
This function does a bit more than comparing both lists. It compares the first `source_dir_size` elements of the lists. Although this seems a bit complicated, I don't think this can be done faster but, if you prefer, we can make it easier to read by creating a sublist and comparing for equality.

Reply:
Ok, I see now.  I think this is the most efficient way now that I understand what it's doing :slightly_smiling_face: